feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/domain/MetaAccountGroupingInteractorImpl.kt

```diff
@@ -125,11 +127,9 @@ class MetaAccountGroupingInteractorImpl(
         )
     }
 
-    private suspend fun getValidMetaAccountsForTransaction(from: Chain, destination: Chain): List<MetaAccount> {
-        val selectedMetaAccount = accountRepository.getSelectedMetaAccount()
-        val fromChainAddress = selectedMetaAccount.addressIn(from)
+    private suspend fun getValidMetaAccountsForTransaction(metaAccountFilter: Filter<MetaAccount>): List<MetaAccount> {
         return accountRepository.allMetaAccounts()
-            .removed { fromChainAddress == it.addressIn(destination) }
```

Comment:
I changed this line to metaAccountFilter to support this feature not only for transfers.
I also found a bug when we use cross chain transfers: meta account was hidden for cross chain transfers in case when `fromChain` and `desitnationChain` has same address prefix. You can see how I fixed this problem [here](https://github.com/novasamatech/nova-wallet-android/pull/1312/files#diff-db66a29c49702694ab93bd4a718ca357c53e2bdc0cd15125fd8265b6dd527443), using metaAccount filter

Reply:
Yeah I remember I noticed this behavior too and was confused about a month ago